meta-client/src/main/java/org/apache/helix/metaclient/impl/zk/ZkMetaClientCache.java

```diff
@@ -81,14 +114,45 @@ public int countDirectChildren(final String key) {
         throw new MetaClientException("Not implemented yet.");
     }
 
-    @Override
-    public List<T> get(List<String> keys) {
-        throw new MetaClientException("Not implemented yet.");
+    private void handleCacheUpdate(String path, ChildChangeListener.ChangeType changeType) {
+        switch (changeType) {
+            case ENTRY_CREATED:
+                // Not implemented yet.
+                break;
+            case ENTRY_DELETED:
+                // Not implemented yet.
+                break;
+            case ENTRY_DATA_CHANGE:
+                modifyDataInCache(path);
+                break;
+            default:
+                LOG.error("Unknown change type: " + changeType);
+        }
     }
 
+    private void modifyDataInCache(String path) {
+        if (_cacheData) {
+            T dataRecord = _cacheClient.readData(path, true);
+            getDataCacheMap().put(path, dataRecord);
+        }
+    }
+
+    public ConcurrentHashMap<String, T> getDataCacheMap() {
+        return _dataCacheMap;
+    }
+
+    public TrieNode getChildrenCacheTree() {
+        return _childrenCacheTree;
+    }
+
+    /**
+     * Connect to the underlying ZkClient.
+     */
     @Override
-    public List<Stat> exists(List<String> keys) {
-        throw new MetaClientException("Not implemented yet.");
+    public void connect() {
+        super.connect();
+        _eventListener = this::handleCacheUpdate;
```

Comment:
Do you think we need a separate thread to handle change events? So that it wont block user requests when we have massive events coming.

Reply:
That's a good point - will add that